### PR TITLE
[KBFS-2843] Save and Share on Android

### PR DESCRIPTION
**Pull request by jzila**

### CODE REVIEW

File: shared/constants/fs.js [hunk 1]
```diff
@@ -287,8 +288,7 @@ export const downloadFilePathFromPathNoSearch = (p: Types.Path): string =>
   downloadFilePathNoSearch(Types.getPathName(p))
 
 export const isImage = (name: string): boolean => {
-  const lower = name.toLowerCase()
-  return ['.png', '.jpg', '.jpeg', '.mp4'].some(ext => lower.endsWith(ext))
+  return (mime.lookup(name) || '').indexOf('image/') !== -1
```
Comment (songgao) [inline shared/constants/fs.js]: Nice! Perhaps include `video/` and `audio/` too? Perhaps this function should've been named `isMedia`.

File: shared/fs/popups/row-action-popup-container.js [hunk 2]
```diff
@@ -115,7 +110,7 @@ const getRootMenuItems = (stateProps, dispatchProps) => {
       onClick: () => saveImage(path),
     })
   // We don't support the share sheet on Android yet.
```
Comment (songgao) [inline shared/fs/popups/row-action-popup-container.js]: Remove this comment?

File: shared/fs/popups/row-action-popup-container.js [hunk 3]
```diff
@@ -33,6 +33,7 @@ const mapStateToProps = (state: TypedState, {routeProps}) => {
         )
 
   return {
+    mimeType: Constants.mimeTypeFromPathItem(pathItem),
```
Comment (songgao) [inline shared/fs/popups/row-action-popup-container.js]: I like naming intermediate props that don't get passed out from `mergeProps` with a `_` prefix. But no big deal.

File: shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/ShareFiles.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    public String getName() {
+        return "ShareFiles";
+    }
+
+    @ReactMethod
+    public void share(String uriPath, String mimeType, Promise promise) {
+        Uri uri = Uri.parse(uriPath);
+        Intent intent = new Intent(Intent.ACTION_SEND)
+                .putExtra(Intent.EXTRA_STREAM, uri)
+                .setType(mimeType);
```
Comment (chrisnojima) [inline shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/ShareFiles.java]: Let's discuss

**Pull request merged**